### PR TITLE
feat: add a globalShortcut to toggle main window

### LINKED ISSUE #1663: Shortcut to Toggle main window
Hello,
I couldn't find if this options is available already.

I would like some way to create a shortcut to Toggle main window to Show/Hide Ferdi quickly!
I use other Electron apps that I can configure it thought a `keymap.cson`:
```
'global':
  'ctrl-shift-x': 'application:toggle-main-window'
```
It minimizes to tray or restore.
This option would improve a lot the productivity. :smile:

### PR DESCRIPTION
**Pull request by michaelhays**
Fixes #1663

- [x] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Add an alt-x [global shortcut](https://www.electronjs.org/docs/latest/api/global-shortcut/) to toggle the main window

#### Motivation and Context
Closes #1663

#### Checklist
- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes
Add an alt-x global shortcut to toggle the main window

### CODE REVIEW

**Comment by michaelhays** [comment]
Let me know if alt-x works for this, or if it would be better to use another shortcut.

Ideally, we'd let the user define the shortcut, but that would depend on #301 being implemented.

File: src/index.js [hunk 1]
```diff
@@ -358,6 +358,13 @@ const createWindow = () => {
   } else {
     mainWindow.show();
   }
+
+  app.whenReady().then(() => {
+    // Toggle the window on 'Alt+X'
+    globalShortcut.register('Alt+X', () => {
```
Comment (vraravam) [inline src/index.js:364]: Please use the shortcut key constant defined in the `environment.ts` file. Please see other usages of the same to pass in the correct parameter.

File: src/index.js [hunk 2]
```diff
@@ -358,6 +358,13 @@ const createWindow = () => {
   } else {
     mainWindow.show();
   }
+
+  app.whenReady().then(() => {
```
Comment (vraravam) [inline src/index.js:362]: Is this supposed to be active only for certain OS? Or is it applicable for all linux, macos and windows?

**Review by vraravam — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (mhatvan) [reply src/index.js:362]: Should work for all OS imo, tested it on Ubuntu, works as expected.

Comment (vraravam) [reply src/index.js:362]: ok - let me resolve this comment

File: src/index.js (hunk 1 above)
Comment (vraravam) [reply src/index.js]: fixed in new commit

**Review by vraravam — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (michaelhays) [reply src/index.js]: Thanks @vraravam!

**Comment by vraravam** [comment]
@ylluminarious - I agree, this should have an option to be turned off. Let me work on it today/tomorrow.

**Comment by ylluminarious** [comment]
@vraravam Thank you, I'll look forward to this in the next nightly release hopefully 🙇

**Comment by vraravam** [comment]
For the immediate need, @ylluminarious - please drop back to the nightly before 25.

@michaelhays - configuring the shortcut can be done if this is exposed in the settings view. Lets take it 1 step at a time: lets first alleviate @ylluminarious 's pain by giving the ability to toggle it on/off. Then we can implement the exposure of the key binding

**Comment by vraravam** [comment]
I just raised a PR (awaiting review) for the preference setting to toggle this on/off. Once it gets merged (or using a stacked PR), please go ahead and implement the user-chosen keyboard shortcut for this feature.

**Comment by ylluminarious** [comment]
> For the immediate need, @ylluminarious - please drop back to the nightly before 25.

@vraravam Thanks for the advice. I tried downgrading to [`5.6.3-beta.1`,](https://github.com/getferdi/ferdi/releases/tag/v5.6.3-beta.1) but apparently that wasn't old enough. I had to go all the way back to [`5.6.3-nightly.19`.](https://github.com/getferdi/ferdi/releases/tag/v5.6.3-nightly.19)

**Pull request merged**